### ISSUE TITLE
CrudAuth persist is called when action is read

### ISSUE THREAD

**Issue by SamiSammour** [nestjsx/crud#466]
I want to use the `persist` function inside the `CrudAuth` decorator to add some data to the create/update/replace operations but the `persist` function is called with every request. I was wondering if we can call it only in these operations and not in the read operation! I can make it in a pull request but I want to know if there's any logic behind this behavior.

One more thing, why don't we make these functions (`filter`, `or` and `persist`) `asynchronous`? I know if I want some complicated logic I should consider using interceptors I think it would be useful to do it like this.

**Comment by zMotivat0r** [comment]
Yep, this seems to be just a bad design from mine. I'll take care of it for sure. Thanks!
